lib/formatters/time_formatter.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Formatters
+  class TimeFormatter
+    ##
+    # format `secs` into a human readable output - X days Y hours Z minutes S seconds
+    #
+    # @param secs [Float|Integer] the number of seconds to format, eg: 940913.38729661
+    #
+    # @return [String] the human readable string
+    #
+    def self.humanize(secs)
+      [[60, :seconds], [60, :minutes], [24, :hours], [Float::INFINITY, :days]].map do |count, name|
+        if secs.positive?
+          secs, n = secs.divmod(count)
+
+          "#{n.to_i} #{name}" unless n.to_i.zero?
```

Comment:
Is there a chance that `secs` will ever equal `0`? if so, what will happen? Or is that covered in this line?

Reply:
There is the chance, if it is a full minute, or 0 is passed - the result, for that iteration, is an empty string:
```
3.3.3 :015 > def humanize(secs)
3.3.3 :016 >   [[60, :seconds], [60, :minutes], [24, :hours], [Float::INFINITY, :days]].map do |count, name|
3.3.3 :017 >     if secs.positive?
3.3.3 :018 >       secs, n = secs.divmod(count)
3.3.3 :019 > 
3.3.3 :020 >       "#{n.to_i} #{name}" unless n.to_i.zero?
3.3.3 :021 >     end
3.3.3 :022 >   end.compact.reverse.join(' ')
3.3.3 :023 > end
 => :humanize 
3.3.3 :024 > 
3.3.3 :025 > humanize 0
 => "" 
3.3.3 :026 > humanize 60
 => "1 minutes" 
3.3.3 :027 > humanize 120
 => "2 minutes" 
3.3.3 :028 > humanize 3600
 => "1 hours" 
3.3.3 :029 > humanize 3660
 => "1 hours 1 minutes" 
3.3.3 :030 > humanize 3780
 => "1 hours 3 minutes" 
3.3.3 :031 >

```

Reply:
ok, and is an empty string acceptable?

Reply:
For what I was using it for yes.  Assumption being that the caller would only do so if >0 and/or want to handle the empty string specific to situation.  I added a note to the docstring